coap/api/transport.go

```diff
@@ -58,50 +46,58 @@ func MakeHandler(instanceID string) http.Handler {
 	return b
 }
 
-// MakeCoAPHandler creates handler for CoAP messages.
-func MakeCoAPHandler(svc coap.Service, channelsClient grpcChannelsV1.ChannelsServiceClient, domainsClient grpcDomainsV1.DomainsServiceClient, l *slog.Logger) mux.HandlerFunc {
-	logger = l
-	service = svc
-	channels = channelsClient
-	domains = domainsClient
+type CoAPHandler struct {
+	logger   *slog.Logger
+	service  coap.Service
+	channels grpcChannelsV1.ChannelsServiceClient
+	resolver messaging.TopicResolver
+}
 
-	return handler
+// MakeCoAPHandler creates handler for CoAP messages.
+func MakeCoAPHandler(svc coap.Service, channelsClient grpcChannelsV1.ChannelsServiceClient, resolver messaging.TopicResolver, l *slog.Logger) mux.HandlerFunc {
```

Comment:
This can be changed to Handler instead of handler func

```suggestion
func MakeCoAPHandler(svc coap.Service, channelsClient grpcChannelsV1.ChannelsServiceClient, resolver messaging.TopicResolver, l *slog.Logger) mux.Handler {
```

Reply:
Created ticket here for this issue , we can solve this later 
https://github.com/absmach/supermq/issues/2985